addons/web_editor/static/src/js/wysiwyg/widgets/media.js

```diff
@@ -1364,55 +1572,162 @@ var VideoWidget = MediaWidget.extend({
+            }
         }
-
-        // Detect if we have an embed code rather than an URL
-        var embedMatch = code.match(/(src|href)=["']?([^"']+)?/);
-        if (embedMatch && embedMatch[2].length > 0 && embedMatch[2].indexOf('instagram')) {
-            embedMatch[1] = embedMatch[2]; // Instagram embed code is different
+        if (!query) {
+            return;
         }
-        var url = embedMatch ? embedMatch[1] : code;
-
-        const query = await this._createVideoNode(url, {
-            'autoplay': this.isForBgVideo || this.$('input#o_video_autoplay').is(':checked'),
-            'hide_controls': this.isForBgVideo || this.$('input#o_video_hide_controls').is(':checked'),
-            'loop': this.isForBgVideo || this.$('input#o_video_loop').is(':checked'),
-            'hide_fullscreen': this.isForBgVideo || this.$('input#o_video_hide_fullscreen').is(':checked'),
-            'hide_yt_logo': this.isForBgVideo || this.$('input#o_video_hide_yt_logo').is(':checked'),
-            'hide_dm_logo': this.isForBgVideo || this.$('input#o_video_hide_dm_logo').is(':checked'),
-            'hide_dm_share': this.isForBgVideo || this.$('input#o_video_hide_dm_share').is(':checked'),
-        });
-
-        var $optBox = this.$('.o_video_dialog_options');
-
+        const $optBox = this.$('.o_video_dialog_options');
         // Show / Hide preview elements
         this.$el.find('.o_video_dialog_preview_text, .media_iframe_video_size').add($optBox).toggleClass('d-none', !query.$video);
-        // Toggle validation classes
-        this.$el.find('#o_video_form_group')
-            .toggleClass('o_has_error', !query.$video).find('.form-control, .custom-select').toggleClass('is-invalid', !query.$video)
-            .end()
-            .toggleClass('o_has_success', !!query.$video).find('.form-control, .custom-select').toggleClass('is-valid', !!query.$video);
 
         // Individually show / hide options base on the video provider
-        $optBox.find('div.o_' + query.platform + '_option').removeClass('d-none');
-
-        // Hide the entire options box if no options are available or if the
-        // dialog is opened for a background-video
-        $optBox.toggleClass('d-none', this.isForBgVideo || $optBox.find('div:not(.d-none)').length === 0);
+        const platformGroup = $optBox.find('div.o_' + query.platform + '_option').removeClass('d-none');
+        const platformOptions = platformGroup.find("input[id^='o_video']");
+        // Gray out and disable options that are non-configurable for bg videos
+        platformOptions.prop("checked", this.isForBgVideo ? true : platformGroup.prop("checked")).prop("disabled", this.isForBgVideo);
+        platformGroup.css("opacity", this.isForBgVideo ? '0.5' : '1');
+        const autoplay = $optBox.find('#o_video_autoplay');
+        autoplay.prop("checked", this.isForBgVideo ? true : autoplay.prop("checked")).prop("disabled", this.isForBgVideo);
+        const loop = $optBox.find('#o_video_loop');
```

Comment:
```suggestion
        const $loop = $optBox.find('#o_video_loop');
```
You got the point, apply everywhere